bee-message/src/payload/fpc/mod.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module describing the FPC statement payload.
+
+mod conflicts;
+mod timestamps;
+
+pub use conflicts::{Conflict, Conflicts};
+pub use timestamps::{Timestamp, Timestamps};
+
+use crate::error::ValidationError;
+
+use bee_packable::{
+    error::{PackPrefixError, UnpackPrefixError},
+    PackError, Packable, Packer, UnpackError, Unpacker,
+};
+
+use core::{convert::Infallible, fmt};
+
+#[derive(Debug)]
+pub enum FpcPackError {
+    InvalidPrefixLength,
+}
+
+impl From<PackPrefixError<Infallible, u32>> for FpcPackError {
+    fn from(error: PackPrefixError<Infallible, u32>) -> Self {
+        match error {
+            PackPrefixError::Packable(e) => match e {},
+            PackPrefixError::Prefix(_) => Self::InvalidPrefixLength,
+        }
+    }
+}
+
+impl fmt::Display for FpcPackError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::InvalidPrefixLength => write!(f, "Invalid prefix length for conflicts/timestamps"),
+        }
+    }
+}
+
+impl From<Infallible> for FpcPackError {
```

Comment:
Maybe we should add a macro inside `bee-packable` that does this.

Reply:
That would be nice!

Reply:
Since there's also a lot of wrapping errors here to provide more information on pack/unpack failure, I was intending on writing a macro in this crate itself to easily wrap a lower-level error, basically to do this quickly:

```rust
impl_wrapped_variant!(PayloadUnpackError::Transaction, TransactionUnpackError);

// Would generate:
impl From<TransactionUnpackError> for PayloadUnpackError {
    fn from(error: TransactionUnpackError) -> Self {
        Self::Transaction(error)
    }
}
```

(But I haven't got around to it yet...)

Reply:
I like it. wen `bee-coercions`?

Reply:
I've implemented this macro now, if you want to take a look 🙂